src/preprocess/visitor.ts

```diff
@@ -0,0 +1,702 @@
+import * as _ from 'lodash';
+import * as escapeStringRegexp from 'escape-string-regexp';
+import * as handlebars from 'handlebars';
+
+import * as yaml from '../yaml';
+import {logger} from '../logger';
+
+import {
+  Env,
+  $EnvValues,
+  AnyButUndefined,
+  ExtendedCfnDoc,
+  GlobalSection,
+  GlobalSectionNames,
+  MaybeStackFrame,
+  validateTemplateParameter,
+  interpolateHandlebarsString,
+  appendPath
+} from './index';
```

Comment:
Let's follow up in another PR with some refactoring to remove the circular dependency between these modules (after I've finished the other module refactoring on cfn/).